include/dts-subscription.sh

```diff
@@ -0,0 +1,341 @@
+#!/usr/bin/env bash
+
+# SPDX-FileCopyrightText: 2024 3mdeb <contact@3mdeb.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# shellcheck disable=SC2034
+# shellcheck source=../include/dts-environment.sh
+source $DTS_ENV
+
+check_for_dasharo_firmware() {
+  # This function checks if Dasharo firmware is available for the current
+  # platform, returns 1 if there is no firmware available, returns 0 otherwise.
+
+  echo "Checking for Dasharo firmware..."
+
+  local _check_dwn_req_resp_uefi="0"
+  local _check_dwn_req_resp_heads="0"
+  local _check_logs_req_resp="0"
+  # Ignore "SC2154 (warning): SE_credential_file is referenced but not assigned"
+  # for external variable:
+  # shellcheck disable=SC2154
+  CLOUDSEND_LOGS_URL=$(sed -n '1p' < ${SE_credential_file} | tr -d '\n')
+  CLOUDSEND_DOWNLOAD_URL=$(sed -n '2p' < ${SE_credential_file} | tr -d '\n')
+  CLOUDSEND_PASSWORD=$(sed -n '3p' < ${SE_credential_file} | tr -d '\n')
+  USER_DETAILS="$CLOUDSEND_DOWNLOAD_URL:$CLOUDSEND_PASSWORD"
+
+  # Check the board information:
+  board_config
+
+  # Create links for logs to be sent to:
+  TEST_LOGS_URL="https://cloud.3mdeb.com/index.php/s/${CLOUDSEND_LOGS_URL}/authenticate/showShare"
+
+  # If board_config function has not set firmware links - exit with warning:
+  if [ ! -v BIOS_LINK_DPP ] && [ ! -v HEADS_LINK_DPP ]; then
+    print_warning "There is no Dasharo Firmware available for your platform."
+    return 1
+  fi
+
+  # Check for firmware binaries:
+  if wait_for_network_connection; then
+    if [ -v BIOS_LINK_DPP ]; then
+      _check_dwn_req_resp_uefi=$(curl -L -I -s -f -u "$USER_DETAILS" -H "$CLOUD_REQUEST" "$BIOS_LINK_DPP" -o /dev/null -w "%{http_code}")
+    fi
+    if [ -v HEADS_LINK_DPP ]; then
+      _check_dwn_req_resp_heads=$(curl -L -I -s -f -u "$USER_DETAILS" -H "$CLOUD_REQUEST" "$HEADS_LINK_DPP" -o /dev/null -w "%{http_code}")
+    fi
+
+    _check_logs_req_resp=$(curl -L -I -s -f -H "$CLOUD_REQUEST" "$TEST_LOGS_URL" -o /dev/null -w "%{http_code}")
+
+    # Return 0 if any of Dasharo Firmware binaries is available:
+    if [ ${_check_dwn_req_resp_uefi} -eq 200 ] || [ ${_check_dwn_req_resp_heads} -eq 200 ]; then
+      if [ ${_check_logs_req_resp} -eq 200 ]; then
+        print_ok "A Dasharo Firmware binary has been found for your platform!"
+        return 0
+      fi
+    fi
+  fi
+
+  print_warning "Something may be wrong with the DPP credentials or you may not\n
+		have access to Dasharo Firmware. If so, consider getting Dasharo\n
+		Subscription and improving security of your platform!"
+
+  read -p "Press ENTER to continue"
+  return 1
+}
+
+get_dpp_creds() {
+  echo ""
+  read -p "Enter logs key:                " 'TMP_CLOUDSEND_LOGS_URL'
+  echo ""
+  read -p "Enter firmware download key:   " 'TMP_CLOUDSEND_DOWNLOAD_URL'
+  echo ""
+  read -p "Enter password:                " 'TMP_CLOUDSEND_PASSWORD'
+
+  echo ${TMP_CLOUDSEND_LOGS_URL} > ${SE_credential_file}
```

Comment:
We use the same credentials file and global variables fro both cloud and minio credentials?
Can we make it clear in the comments, that this is the case, and what the mapping between these variables is?

Reply:
> We use the same credentials file and global variables fro both cloud and minio credentials?

Yep.

Reply:
Added comments here 6e2ba4b9caaefe8e7cf12a9247479486818efcde.